test/UnitTest/Microsoft.AspNet.OData.Test.Shared/Formatter/Deserialization/ODataPrimitiveDeserializerTests.cs

```diff
@@ -199,6 +236,93 @@ public void Read_Primitive(object obj, string edmType, string value)
             Assert.Equal(obj, deserializer.Read(messageReader, type, readContext));
         }
 
+        [Theory]
+        [InlineData("{\"value\":\"FgQF\"}", typeof(byte[]), new byte[] { 22, 4, 5 })]
+        [InlineData("{\"value\":1}", typeof(Int16), (Int16)1)]
+        [InlineData("{\"value\":1}", typeof(Int32), 1)]
+        [InlineData("{\"value\":1}", typeof(Int64), (Int64)1)]
+        [InlineData("{\"value\":\"true\"}", typeof(Boolean), true)]
+        [InlineData("{\"value\":5}", typeof(SByte), (SByte)5)]
+        [InlineData("{\"value\":201}", typeof(Byte), (Byte)201)]
+        [InlineData("{\"value\":1.1}", typeof(Double), 1.1)]
+        [InlineData("{\"value\":1.1}", typeof(Single), (Single)1.1)]
+        public void ReadFromStreamAsync_RawPrimitive(string content, Type type, object expected)
+        {
+            // Arrange
+            IEdmModel model = CreateModel();
+
+            ODataPrimitiveDeserializer deserializer = new ODataPrimitiveDeserializer();
+            ODataDeserializerContext readContext = new ODataDeserializerContext
+            {
+                Model = model,
+                ResourceType = type
+            };
+            
+            // Act
+            object value = deserializer.Read(ODataDeserializationTestsCommon.GetODataMessageReader(ODataDeserializationTestsCommon.GetODataMessage(content, new HttpRequestMessage(new HttpMethod("Patch"), "http://localhost/OData/Suppliers(1)/Address")), model), type, readContext);
+
+            // Assert
+            Assert.Equal(expected,value);
+        }
+
+        [Fact]
+        public void ReadFromStreamAsync_RawGuid()
```

Comment:
>ReadFromStreamAsync_RawGuid [](start = 20, length = 27)

Guid can't test in above "Test method"?, like other types (single, double)?

Reply:
Guid is different compared to other simple primitive types as new Guid() or Guid.NewGuid is not treated as a constant.